### PR TITLE
fix(tests): code report, git file diff. Refactor 'pylint' module. Refactor tests.

### PR DESCRIPTION
**Pull request by i-savynikh**
Fix link to 'python3' in docker image which used for 'code report' tests. Cause: 'python3' used to install 'pylint' by universum tests and 'python3' used to run 'pylint' aren't same.
Refactor tests for 'argument_check', 'test_api', partially refactor tests for 'check_if_env_set'.

### CODE REVIEW

File: analyzers/pylint.py [hunk 1]
```diff
@@ -25,61 +24,62 @@ class PylintAnalyzer:
...
                             help="Version of Python")
         utils.add_common_arguments(parser)
         return parser
 
     def __init__(self, settings):
         self.settings = settings
-        self.json_file = settings.result_file
 
     def execute(self):
-        if not self.settings.file_list:
```
Comment (miltolstoy) [inline analyzers/pylint.py]: Why you dropped this check?

File: analyzers/utils.py [hunk 2]
```diff
@@ -12,10 +12,9 @@ def add_common_arguments(parser):
                              "script separately from Universum, just name the result file or leave it empty.")
 
 
-def analyzers_output(json_file, issues_loads):
+def analyzers_output(json_file: str, issues_loads) -> None:
     issues = json.dumps(issues_loads, indent=4)
-    if not json_file:
-        sys.stdout.write(issues)
+    if json_file:
+        open(json_file, "w").write(issues)
```
Comment (miltolstoy) [inline analyzers/utils.py]: File not closed.

File: tests/configs.py [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+
 import os
 from _universum.configuration_support import Variations
 
 env_name = "virtual_universe"
 
 
 def run_virtual(cmd):
-    return ["env", "-i", "PATH=" + os.getenv("PATH"), "bash", "-c", "source {}/bin/activate; {}".format(env_name, cmd)]
+    return ["env", "-i", "PATH=" + os.getenv("PATH"), "bash", "-c", f"source {env_name}/bin/activate; {cmd}"]
```
Comment (k-dovgan) [inline tests/configs.py]: 20:55:34  2. Processing project configs
20:55:34   |   Error: Exception while processing 'configs.py'
20:55:34   |   Traceback (most recent call last):
20:55:34   |     File "/usr/local/lib/python2.7/dist-packages/_universum/modules/launcher.py", line 273, in process_project_configs
20:55:34   |       exec(config.read(), config_globals)  # pylint: disable=exec-used
20:55:34   |     File "<string>", line 9
20:55:34   |       return ["env", "-i", "PATH=" + os.getenv("PATH"), "bash", "-c", f"source {env_name}/bin/activate; {cmd}"]
20:55:34   |                                                                                                              ^
20:55:34   |   SyntaxError: invalid syntax

File: tests/test_code_report.py [hunk 4]
```diff
@@ -1,48 +1,83 @@
...
 from _universum.configuration_support import Variations
 
-configs = Variations([dict(name="Run static pylint", code_report=True,
-                           command=["universum_pylint", "--python-version=3", "--files", "source_file.py",
-                           "--result-file", "${CODE_REPORT_FILE}"])])
-"""
+configs = Variations([dict(name="Run static pylint", code_report=True, command=['universum_pylint'"""
+    for arg in args:
+        config += f", '{arg}'"
+    return f"{config}])])"
```
Comment (k-dovgan) [inline tests/test_code_report.py]: Why is this needed? It's only called once with one set of arguments, isn't it?

File: analyzers/pylint.py (hunk 1 above)
Comment (i-savynikh) [reply analyzers/pylint.py]: because I add 'required=True' to parameter

File: analyzers/utils.py (hunk 2 above)
Comment (i-savynikh) [reply analyzers/utils.py]: python close file automatically when out of context

File: tests/configs.py (hunk 3 above)
Comment (i-savynikh) [reply tests/configs.py]: please use python3.7 instead

File: tests/test_code_report.py (hunk 4 above)
Comment (i-savynikh) [reply tests/test_code_report.py]: not once

File: _universum/modules/vcs/git_vcs.py [hunk 5]
```diff
@@ -148,6 +148,11 @@ def cherry_pick(self):
         self.append_repo_status("\n")
 
     def _diff_against_reference_commit(self, commit_id):
+        """Details. Depends on a 'git' version 'rename file' operation generates
```
Comment (i-keliukh) [inline _universum/modules/vcs/git_vcs.py]: Depends -> Depending

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,39 @@
+astroid==2.3.3
```
Comment (i-keliukh) [inline requirements.txt]: Why we switch to requirements.txt?
What are the consequences?

File: tests/configs.py (hunk 3 above)
Comment (k-dovgan) [inline tests/configs.py]: Please remove fstrings from this file until final switching to Python3.7; instead please add 'python3.7 -m' into Variations

File: Makefile [hunk 7]
```diff
@@ -20,15 +20,15 @@ test:
 	for t in $(TEST_TARGETS); do $(MAKE) $$t || error=1; done; exit $$error
 
 pytest:
-	python2 -m pytest --doctest-modules -vv --junitxml=junit_results.xml --cov-report=html \
+	python3 -m pytest --doctest-modules -vv --junitxml=junit_results.xml --cov-report=html \
```
Comment (k-dovgan) [inline Makefile]: Python3 is Python3.5, that's why the build still fails. All required packages are installed under pip3.7 (presumably)
